pyproject.toml

```diff
@@ -39,7 +39,7 @@ mypy = "1.9.0"
 pandas = "2.2.1"
 pyarrow = ">=10.0.1"
 pytest = ">=7.1.2"
-pyright = ">=1.1.354"
+pyright = "=1.1.354"
```

Comment:
#897 fixed this issue, so you can use `pyright = ">=1.1.357"` and CI should pass